utils/src/main/kotlin/Logger.kt

```diff
@@ -39,7 +39,14 @@ val loggerOfClass = ConcurrentHashMap<Any, KotlinLogger>()
  * An extension property for adding a log instance to any (unique) class.
  */
 val <reified T : Any> T.log: KotlinLogger
-    inline get() = loggerOfClass.getOrPut(T::class.java) { loggerOf(T::class.java) }
+    inline get() = loggerOfClass.getOrPut(T::class.java) {
+        val name = T::class.qualifiedName.orEmpty()
+        require(name.startsWith("org.ossreviewtoolkit")) {
```

Comment:
The `require()` statement could be problematic IMHO. As the problem is not discovered at compile time, there is a certain chance that you get unexpected runtime exceptions if certain log statements are not covered by tests. So would it make sense to log a warning statement with a well-known standard logger?

How do the class names for these lambda classes look like? Would it be somehow possible to map them to the ORT classes they are contained in?

Reply:
> So would it make sense to log a warning statement with a well-known standard logger?

I fear that warnings in the log tend to be overlooked if you're not specifically looking for them.

> How do the class names for these lambda classes look like?

Actually, my initial comment that it would help to identify lambdas was bogus. At this point we just see the class the lambda runs against, not the lambda itself.

Reply:
So IIUC, this would catch statements like `42.log { "Log from an Int" }`, but would not help us to detect the problems #3608 is about, right?

Reply:
IMO it *would* detect that misuse, as `log` is run on an instance of `java.sql.PreparedStatement`.

Reply:
I like that idea, would you mind adding a test case?

Reply:
I've adjusted the existing https://github.com/oss-review-toolkit/ort/blob/master/utils/src/test/kotlin/LoggerTest.kt.